src/components/sbb-checkbox/sbb-checkbox.tsx

```diff
@@ -24,170 +10,195 @@ import {
       this._requiredFromGroup = isValidAttribute(parentGroup, 'required');
       this._disabledFromGroup = isValidAttribute(parentGroup, 'disabled');
       this.size = parentGroup.size;
     }
-    this._checkboxAttributeObserver.observe(this._element, checkboxObserverConfig);
+    this._checkboxAttributeObserver.observe(this, checkboxObserverConfig);
   }
 
   /** Observe changes on data attributes and set the appropriate values. */
   private _onCheckboxAttributesChange(mutationsList: MutationRecord[]): void {
     for (const mutation of mutationsList) {
       if (mutation.attributeName === 'data-group-disabled') {
-        this._disabledFromGroup = !!isValidAttribute(this._element, 'data-group-disabled');
+        this._disabledFromGroup = !!isValidAttribute(this, 'data-group-disabled');
       }
       if (mutation.attributeName === 'data-group-required') {
-        this._requiredFromGroup = !!isValidAttribute(this._element, 'data-group-required');
+        this._requiredFromGroup = !!isValidAttribute(this, 'data-group-required');
       }
     }
   }
 
-  public connectedCallback(): void {
+  public override connectedCallback(): void {
+    super.connectedCallback();
+    const signal = this._abort.signal;
+    this.addEventListener('click', (e) => this._handleClick(e), { signal });
+    this.addEventListener('keyup', (e) => this._handleKeyup(e), { signal });
     // We can use closest here, as we expect the parent sbb-selection-panel to be in light DOM.
-    this._selectionPanelElement = this._element.closest('sbb-selection-panel');
+    this._selectionPanelElement = this.closest('sbb-selection-panel');
     this._isSelectionPanelInput =
-      !!this._selectionPanelElement &&
-      !this._element.closest('sbb-selection-panel [slot="content"]');
+      !!this._selectionPanelElement && !this.closest('sbb-selection-panel [slot="content"]');
     this._handlerRepository.connect();
     this._setupInitialStateAndAttributeObserver();
   }
 
-  public componentDidLoad(): void {
-    this._isSelectionPanelInput && this._updateExpandedLabel();
+  public override willUpdate(changedProperties: PropertyValues<this>): void {
+    if (changedProperties.has('checked')) {
+      this._handleCheckedChange(this.checked, changedProperties.get('checked'));
+    }
+    if (changedProperties.has('disabled')) {
+      this._handleDisabledChange(this.disabled, changedProperties.get('disabled'));
+    }
+  }
+
+  protected override firstUpdated(): void {
+    this._selectionPanelElement && this._updateExpandedLabel();
   }
 
-  public disconnectedCallback(): void {
+  public override disconnectedCallback(): void {
+    super.disconnectedCallback();
     this._handlerRepository.disconnect();
     this._checkboxAttributeObserver.disconnect();
   }
 
-  @Listen('click')
-  public handleClick(event: Event): void {
-    if (!this.disabled && !this._disabledFromGroup && getEventTarget(event) === this._element) {
-      this._checkbox.click();
```

Comment:
Does this really not work anymore?

Reply:
From what i see, it works only in tests; if i comment it, the component keeps working in storybook but tests fail.

Reply:
it's ok, was just unexpected :D